### PR TITLE
feat(signin-code): Add bounce handling to /signin_token_code

### LINKED ISSUE #3528: /signin_token_code does not handle bounces
Similar to #3405, /signin_token_code does not handle email bounces.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
If an email bounces when sending a token code, we never knew and
the user was not informed. This sends the user to /signin_bounced.

fixes #3528

@vbudhram - r?

I wasn't sure if there needed to be any additional logic on the back-end.

### CODE REVIEW

File: packages/fxa-content-server/app/scripts/views/sign_in_token_code.js [hunk 1]
```diff
@@ -29,6 +30,12 @@ const View = FormView.extend({
     }
   },
 
+  afterVisible() {
+    this.waitForSessionVerification(this.getAccount(), () => {
```
Comment (vbudhram) [inline packages/fxa-content-server/app/scripts/views/sign_in_token_code.js]: I would probably add a quick comment about why we need to poll for verification here.

Comment (vbudhram) [reply packages/fxa-content-server/app/scripts/views/sign_in_token_code.js]: Conversely, we could add a new helper function `pollForEmailBounce` or something similar, but meh.

File: packages/fxa-content-server/tests/functional/sign_in_token_code.js [hunk 2]
```diff
@@ -103,6 +105,22 @@ registerSuite('signin token code', {
...
+    'verified - treatment-code - bounce': function() {
+      experimentParams.query.forceExperiment = 'tokenCode';
+      experimentParams.query.forceExperimentGroup = 'treatment-code';
+
+      return this.remote
+        .then(openFxaFromRp('enter-email', experimentParams))
+        .then(fillOutEmailFirstSignIn(email, PASSWORD))
+        .then(testElementExists(selectors.SIGNIN_TOKEN_CODE.HEADER))
+        .then(getStoredAccountByEmail(email))
+        .then(destroySessionForEmail(email))
```
Comment (vbudhram) [inline packages/fxa-content-server/tests/functional/sign_in_token_code.js]: While looking at https://github.com/mozilla/fxa/pull/3533/files, why would this have to use `destroySessionForEmail` verses `accountDestroy`?

Comment (shane-tomlinson) [reply packages/fxa-content-server/tests/functional/sign_in_token_code.js]: A sign in verification email doesn't normally destroy the account, that only happens on signup verifications.

**Comment by vasilicatamas** [comment]
Some reliable STR would be very helpful, if the QA attention is still needed.

**Pull request merged**